src/tsolvers/egraph/EgraphSolver.cc

```diff
@@ -269,8 +269,10 @@ void Egraph::declareTermRecursively(PTRef tr) {
     if (declared.find(tr) != declared.end()) { return; } // already declared
     const Pterm& term = logic.getPterm(tr);
     // declare first the childen and then the current term
-    for (int i = 0; i < term.size(); ++i) {
-        declareTermRecursively(term[i]);
+    if (not logic.isNonBoolIte(tr)) {
```

Comment:
Also here, it seems like boolean and non-boolean ITE should now be treated the same.